ADDON_HOOKS.md

```diff
@@ -429,15 +429,32 @@ Addon.prototype.treeFor = function treeFor(name) {
 <a name='treefor-cont'></a>
 # treeFor (cont...)
 
-Instead of overriding `treeFor` and acting only if the tree you receive matches the one you need EmberCLI has custom hooks for the following Broccoli trees
+Instead of overriding `treeFor` and acting only if the tree you receive matches the one you need EmberCLI has custom hooks for the following Broccoli trees:
 
 - treeForApp
 - treeForStyles
 - treeForTemplates
-- treeForAddon
+- treeForAddon*
 - treeForVendor
 - treeForTestSupport
-- treeForPublic
+- treeForPublic*
+
+_*Denotes a hook that has a default implementation. If you implement a custom function, you may want to call `_super` in order to preserve exisiting functionality._
```

Comment:
rule of thumb: When overriding a frameworking provided method, if in any branch you want to do the "default", even if that is a noop. Call super.